File: Selenide/Chapter1-Selenide+Android(Native)OnSingleDevice/src/test/java/org/selenide/examples/pcloudy/singleDevice.java
  package org.selenide.examples.pcloudy;

  import com.codeborne.selenide.Configuration;
  import com.codeborne.selenide.WebDriverRunner;
  import org.junit.jupiter.api.BeforeEach;
  import org.junit.jupiter.api.Test;
  import org.openqa.selenium.By;
  import org.openqa.selenium.support.events.AbstractWebDriverEventListener;

  import static com.codeborne.selenide.Condition.text;
  import static com.codeborne.selenide.Selenide.$;
  import static com.codeborne.selenide.Selenide.closeWebDriver;
  import static com.codeborne.selenide.Selenide.open;
  import java.io.*;

  
  public class singleDevice {

    @BeforeEach
    void setUp() {
      closeWebDriver();
      Configuration.browserSize = null;
      Configuration.browser = pcloudyconnection.class.getName();
      WebDriverRunner.addListener(new AbstractWebDriverEventListener() {
      });

      open();
    }

    @Test
    void DemoApp() {
      $(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/accept']"))
      .click();
      System.out.println("Accept Button clicked");

      $(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/flightButton']"))
      .click();
      System.out.println("Flight Button clicked");

      $(By.xpath("//android.widget.Spinner[@resource-id='com.pcloudy.appiumdemo:id/spinnerfrom']"))
      .click();
      System.out.println("from selected");

      $(By.xpath("//android.widget.CheckedTextView[@resource-id='android:id/text1' and @text='Bangalore, India (BLR)']"))
      .click();
      System.out.println("From city selected");

      $(By.xpath("//android.widget.Spinner[@resource-id='com.pcloudy.appiumdemo:id/spinnerto']"))
      .click();
      System.out.println("To selected");

      $(By.xpath("//android.widget.CheckedTextView[@resource-id='android:id/text1' and @text='Pune, India (PNQ)']"))
      .click();
      System.out.println("To city selected");

      $(By.xpath("//android.widget.RadioButton[@resource-id='com.pcloudy.appiumdemo:id/singleTrip']"))
      .click();
      System.out.println("Single Trip Selected");

      $(By.xpath("//android.widget.TextView[@resource-id='com.pcloudy.appiumdemo:id/txtdepart']"))
      .click();
      System.out.println("Depart selected");

      $(By.xpath("//android.widget.Button[@resource-id='android:id/button1' and @text='OK']"))
      .click();
      System.out.println("Ok clicked");

      $(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/searchFlights']"))
      .click();
      System.out.println("Search flight Button clicked");
    }
  }

File: Mocha/Chapter1-Mocha+Android(Native)OnSingleDevice/specs/singleDevice.js
            const { default: driver } = require("appium-android-driver/build/lib/driver");
            const { By } = require("selenium-webdriver");
            const { capabilities } = require("../conf/singleDevice.conf");

            require("appium-android-driver")
            var assert= require("assert"),
            webdriver = require("selenium-webdriver"),
            conf_file= process.argv[3];
            

            var caps = require("../" + conf_file).capabilities;

            var buildDriver = function(caps) {
            return new webdriver.Builder()
                .usingServer(
                "https://device.pcloudy.com/appiumcloud/wd/hub"
                )
                .withCapabilities(caps)
                .build();
            };



            describe("Mocha Demo " + caps.browserName, function() {
            var driver;
            this.timeout(0);
            
            
                it ('Application is launched', function name(done) { 
                    driver=buildDriver(caps);

                    
                
                driver.findElement(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/accept']"))
                .click().then(function(){
                console.log("Accept Button clicked");
                });

                driver.findElement(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/flightButton']"))
                .click().then(function(){
                console.log("Flight Button clicked");
                });

                driver.findElement(By.xpath("//android.widget.Spinner[@resource-id='com.pcloudy.appiumdemo:id/spinnerfrom']"))
                .click().then(function(){
                console.log("from selected");
                });

                driver.findElement(By.xpath("//android.widget.CheckedTextView[@resource-id='android:id/text1' and @text='Bangalore, India (BLR)']"))
                .click().then(function(){
                console.log("From city selected");
                });

                driver.findElement(By.xpath("//android.widget.Spinner[@resource-id='com.pcloudy.appiumdemo:id/spinnerto']"))
                .click().then(function(){
                console.log("To selected");
                });

                driver.findElement(By.xpath("//android.widget.CheckedTextView[@resource-id='android:id/text1' and @text='Pune, India (PNQ)']"))
                .click().then(function(){
                console.log("To city selected");
                });

                driver.findElement(By.xpath("//android.widget.RadioButton[@resource-id='com.pcloudy.appiumdemo:id/singleTrip']"))
                .click().then(function(){
                console.log("Single Trip Selected");
                });

                driver.findElement(By.xpath("//android.widget.TextView[@resource-id='com.pcloudy.appiumdemo:id/txtdepart']"))
                .click().then(function(){
                console.log("Depart selected");
                });

                driver.findElement(By.xpath("//android.widget.Button[@resource-id='android:id/button1' and @text='OK']"))
                .click().then(function(){
                console.log("Ok clicked");
                });

                driver.findElement(By.xpath("//android.widget.Button[@resource-id='com.pcloudy.appiumdemo:id/searchFlights']"))
                .click().then(function(){
                console.log("Search flight Button clicked");
                });
                driver.quit().then(function(){
                    done();
                 });
            });
            });
            

